tests/e2e/collectives/CMakeLists.txt

```diff
@@ -0,0 +1,42 @@
+# Copyright 2023 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+# These tests perform linking via the Compiler API, which is only supported
+# in bundled-LLVM builds at the moment (#14086).
+if(NOT IREE_BUILD_BUNDLED_LLVM)
+  return()
+endif()
+
+if(IREE_TARGET_BACKEND_CUDA AND IREE_HAL_DRIVER_CUDA)
+  iree_py_test(
+    NAME
+      collectives_test_single_rank
+    SRCS
+      "collectives_test.py"
+    ARGS
+      "-k" "SingleRank"
+      "--target_backend=cuda"
+      "--driver=cuda"
+    LABELS
+      "requires-gpu-nvidia"
+      "driver=cuda"
+  )
+
+  iree_py_test(
+    NAME
+      collectives_test_two_ranks
+    SRCS
+      "collectives_test.py"
+    ARGS
+      "-k" "TwoRanks"
+      "--target_backend=cuda"
+      "--driver=cuda"
+    LABELS
+      "requires-gpu-nvidia"
+      "driver=cuda"
+      "device-count=2"
```

Comment:
> How does this test behave on machines with only one GPU device?

If there are less than 2 devices it will fail during initialization with an error coming from NCCL.

We can have `IREE_ENABLE_MULTIGPU_TESTS`, but I thought that test labels are useful for just that case.

> I'm wondering what options are available here to feature detect at configure/build/run-time

The configuration whether to run a test should be pragmatically simple to avoid situations where a test that needs to run is by mistake disabled due to a programming error in the test configuration code.

Reply:
The test labels do cover that case, but we'd still like for as many tests as possible to pass when just using `ctest` without any arguments (https://openxla.github.io/iree/building-from-source/getting-started/#running-tests)

It's probably fine here without adding an extra CMake variable, at least if the tests have a descriptive enough name that people can see the failures and understand reasonably quickly that they are expected to fail on their setups. On that note, how do you feel about putting "multi_gpu" in the test name, or something similar?

Reply:
I renamed the tests to `collectives_test_single_gpu` and `collectives_test_multi_gpu`.